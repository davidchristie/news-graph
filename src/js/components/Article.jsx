import React, { PropTypes } from 'react'

export default class Article extends React.Component {
  render() {
    const article = this.props.article
    const url = article.url
    return (
      <div className="card col-md-6 col-lg-4">
        <a href={url} className="embedly-card"></a>
        <p>
          Some details about the article.
        </p>
      </div>
    )
  }
}

Article.propTypes = {
  article: PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired
}
