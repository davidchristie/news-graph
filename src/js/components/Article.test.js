/* eslint-disable no-undef */

import React from 'react'
import renderer from 'react-test-renderer'

import Article from './Article'

const exampleArticle = {
  id: 1,
  url: 'http://www.nature.com/news/ancient-retroviruses-emerged-half-a-billion-years-ago-1.21274'
}

test('Article renders correctly', () => {
  const tree = renderer.create(
    <Article article={exampleArticle} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
