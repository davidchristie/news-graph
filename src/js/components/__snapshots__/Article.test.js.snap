exports[`test Article renders correctly 1`] = `
<div
  className="card col-md-6 col-lg-4">
  <a
    className="embedly-card"
    href="http://www.nature.com/news/ancient-retroviruses-emerged-half-a-billion-years-ago-1.21274" />
  <p>
    Some details about the article.
  </p>
</div>
`;
